jetty-start/src/main/java/org/eclipse/jetty/start/CommandLineBuilder.java

```diff
@@ -53,87 +54,155 @@ public static String findJavaBin()
         return "java";
     }
 
-    /**
-     * Perform an optional quoting of the argument, being intelligent with spaces and quotes as needed. If a subString is set in quotes it won't the subString
-     * won't be escaped.
-     *
-     * @param arg the argument to quote
-     * @return the quoted and escaped argument
-     * @deprecated no replacement, quoting is done by {@link #toQuotedString()} now.
-     */
-    @Deprecated
-    public static String quote(String arg)
+    private final StringBuilder commandLine = new StringBuilder();
+    private final List<String> args = new ArrayList<>();
+    private final String separator;
+
+    public CommandLineBuilder()
     {
-        return "'" + arg + "'";
+        this(false);
     }
 
-    private List<String> args;
-
-    public CommandLineBuilder()
+    public CommandLineBuilder(boolean multiline)
     {
-        args = new ArrayList<String>();
+        separator = multiline ? (" \\" + System.lineSeparator() + "  ") : " ";
     }
 
-    public CommandLineBuilder(String bin)
+    /**
+     * Quote a string suitable for use with a command line shell using double quotes.
+     * <p>This method applies doubles quoting as described for the unix {@code sh} commands:
+     * Enclosing characters within double quotes preserves the literal meaning of all characters except
+     * dollarsign ($), backquote (`), and backslash (\).
+     * The backslash inside double quotes is historically weird, and serves
+     * to quote only the following characters: {@code $ ` " \ newline}.
+     * Otherwise, it remains literal.
```

Comment:
```suggestion
     * Quote a string suitable for use with a command line shell using strong quotes {@code '}.
     * <p>This method applies strong quoting as described for the unix {@code sh} commands:
     * Enclosing characters within strong quotes preserves the literal meaning of all characters.
```

We don't want to use double quotes with sh / bash / zsh, we want to use strong quotes `'`, as that will not interpret the contents between the quotes.

Examples:

**`sh` on ubuntu**

```
[joakim@hyperion ~]$ which sh
/usr/bin/sh
[joakim@hyperion ~]$ ls -la /usr/bin/sh
lrwxrwxrwx 1 root root 4 Mar 23  2022 /usr/bin/sh -> dash
[joakim@hyperion ~]$ sh -i -l
$ echo "Hello $USER"
Hello joakim
$ echo 'Hello $USER'
Hello $USER
```

**`bash` on ubuntu**

```
[joakim@hyperion ~]$ echo "Hello $USER"
Hello joakim
[joakim@hyperion ~]$ echo 'Hello $USER'
Hello $USER
```

**`zsh` on ubuntu**

```
[joakim@hyperion ~]$ zsh -i -l
joakim@hyperion ~ % echo "Hello $USER"
Hello joakim
joakim@hyperion ~ % echo 'Hello $USER'
Hello $USER
```

Reply:
Using single quotes is ugly/complex because we have to escape literal single quotes outside of the scope of the quotes, so quoting something like `my.property='foo bar'` (where the single quotes are literal) will end up with the unreadable `my.property=\''foo bar'\'` 

Double quotes a simpler and with 4 characters that need escaping are well enough defined and simple to support.

Reply:
The example property `my.property='foo bar'` would, as if it appeared in `start.d/my.ini` would mean that the end user wants the single quotes in the value as the end result (meaning if they do `properties.get("my.property")` they would expect a String with those single-tick characters: `"'foo bar'"`)

The end result would be for strong quotes ...
```
'-Dmy.property='\''foo bar'\'''
```

The entire argument is strong quoted, not just a section.

Example:

```
[joakim@hyperion ~]$ java -Dmy.property='foo bar' -XshowSettings:properties --version 2>&1 | grep my.prop
    my.property = foo bar
[joakim@hyperion ~]$ java '-Dmy.property='\''foo bar'\''' -XshowSettings:properties --version 2>&1 | grep my.prop
    my.property = 'foo bar'
```

This Strong Quoting also eliminates ALL interpretation of the contents of the quoted string by the shell (on all shells)

From the GNU dash man page (what the standard `sh` is on Ubuntu)

>   **Single Quotes**
>     Enclosing characters in single quotes preserves the literal meaning
>     of all the characters (except single quotes, making it impossible
>     to put single-quotes in a single-quoted string).

Something that Double Quotes does not do, and escaping will need to be SUPER robust (things you are ignoring in this PR will fail on various shells) if we go that route.

Yes, I'm aware that `dash` only mentions a handful of characters for Double Quotes, the other shells have far more special characters when encountered within Double Quotes (a behavior that Strong Quoting doesn't have)

Reply:
@joakime I think what you have been trying to say is: "xargs doesn't grok double quotes like y'd think".  Once I worked that out, then OK let's go with single quotes.   updated.

Reply:
> @joakime I think what you have been trying to say is: "xargs doesn't grok double quotes like y'd think". Once I worked that out, then OK let's go with single quotes. updated.

@joakime what I meant to say was... thanks for persisting and I understood the issue eventually.

Reply:
> @joakime I think what you have been trying to say is: "xargs doesn't grok double quotes like y'd think". Once I worked that out, then OK let's go with single quotes. updated.

It's not xargs doing that. 
It is `bash` doing that, `bash` (and `zsh`) will evaluate on any redirection/pipe before redirection/pipe (unlike `dash`).
This was a design decision by `bash` to get around deficiencies in `sh` not allowing variables to be set on redirection/pipe.